docs/preview/yamlgettingstarted-templates.md

```diff
@@ -0,0 +1,295 @@
+# YAML getting started - Templates (not available, for discussion only)
+
+## Example: Simple input
+
+The following example illustrates two things:
+
+1. Front-matter is used to define the default values for inputs.
+
+2. Inputs are expanded using template expressions. A value that starts and ends
+   with `${{ }}` indicates a template expression.
+
+Template:
+
+```yaml
+---
+# Default values
+inputs:
+  solution: '**/*.sln'
+---
+steps:
+- task: msbuild@1
```

Comment:
Can templates be nested?

Reply:
We will probably start with nested, as long as the template is for a lower-level object. So for instance, a template that contains phases can in-turn reference templates that contain steps.

And one other thing to point out, templates return a list of things. So if a template is referenced at the phases level, it can return one or more phases (or zero).